File: CS/ImageMemoEdit/ExtendedEditor/MyMemoEditPainter.cs
using System;
using System.Collections.Generic;
using System.Linq;
using DevExpress.XtraEditors.Drawing;
using System.Drawing;

namespace EditorDescendant
{
    public class MyMemoEditPainter : MemoEditPainter
    {

        public MyMemoEditPainter()
        {

        }

        public override void Draw(ControlGraphicsInfoArgs info)
        {
            info.Cache.FillRectangle(Brushes.White, info.Bounds);
            base.Draw(info);
        }

        protected override void DrawContent(ControlGraphicsInfoArgs info)
        {
            base.DrawContent(info);
            DrawIcon(info);
        }

        protected virtual void DrawIcon(ControlGraphicsInfoArgs info)
        {
            MyMemoEditViewInfo vi = info.ViewInfo as MyMemoEditViewInfo;

            Image img = vi.GetImage();
            if (img != null)
            {
                Rectangle rec = new Rectangle(info.Bounds.X + vi.IconRect.X, info.Bounds.Y + vi.IconRect.Y, vi.IconRect.Width, vi.IconRect.Height);
                info.Graphics.DrawImage(img, rec);
            }
            else
                info.Graphics.FillRectangle(info.Cache.GetSolidBrush(Color.White), vi.IconRect);
        }
    }
}

File: VB/ImageMemoEdit/ExtendedEditor/MyMemoEditPainter.vb
Imports DevExpress.XtraEditors.Drawing
Imports System.Drawing

Namespace EditorDescendant

    Public Class MyMemoEditPainter
        Inherits MemoEditPainter

        Public Sub New()
        End Sub

        Public Overrides Sub Draw(ByVal info As ControlGraphicsInfoArgs)
            info.Cache.FillRectangle(Brushes.White, info.Bounds)
            MyBase.Draw(info)
        End Sub

        Protected Overrides Sub DrawContent(ByVal info As ControlGraphicsInfoArgs)
            MyBase.DrawContent(info)
            DrawIcon(info)
        End Sub

        Protected Overridable Sub DrawIcon(ByVal info As ControlGraphicsInfoArgs)
            Dim vi As MyMemoEditViewInfo = TryCast(info.ViewInfo, MyMemoEditViewInfo)
            Dim img As Image = vi.GetImage()
            If img IsNot Nothing Then
                Dim rec As Rectangle = New Rectangle(info.Bounds.X + vi.IconRect.X, info.Bounds.Y + vi.IconRect.Y, vi.IconRect.Width, vi.IconRect.Height)
                info.Graphics.DrawImage(img, rec)
            Else
                info.Graphics.FillRectangle(info.Cache.GetSolidBrush(Color.White), vi.IconRect)
            End If
        End Sub
    End Class
End Namespace
